src/realm/query_engine.cpp

```diff
@@ -684,3 +686,74 @@ ExpressionNode::ExpressionNode(const ExpressionNode& from, QueryNodeHandoverPatc
 , m_expression(from.m_expression->clone(patches))
 {
 }
+
+ReadAccessNode::ReadAccessNode(size_t acl_column_index, StringData user_id)
+    : m_acl_column(acl_column_index)
+    , m_user_id(user_id)
+{
+    m_dD = 10.0;
+    m_dT = 50.0;
+}
+
+void ReadAccessNode::table_changed()
+{
+    REALM_ASSERT(m_table->get_column_type(m_acl_column) == type_LinkList);
+    m_column = &m_table->get_column_link_list(m_acl_column);
+
+    m_permissions = const_cast<Table*>(m_table->get_link_target(m_acl_column).get());
+
+    m_read_col_ndx = m_permissions->get_column_index("canRead");
```

Comment:
Use of constant strings "canRead", "role" and "members" appear to be specifically targeted at Sync's implementation of access control. Is this the right thing to do in Core?